### PR TITLE
[skip ci] Doctest for Loss

### LINKED ISSUE #2265: [docs] Add doctests for metrics
### 📚 Documentation

Following #2230 , I listed the metrics places (suggested by @sdesrozis) where doctests can be added.

#### Metrics

- [x] Average (#2332)
- [x] GeometricAverage (#2332)
- [x] VariableAccumulation (This class is an helper, no test needed)
- [x] Accuracy (#2345)
- [x] ConfusionMatrix (#2336)
- [x] ClassificationReport (#2346)
- [x] DiceCoefficient (#2336)
- [x] JaccardIndex (#2336)
- [x] IoU (#2336)
- [x] mIoU (#2336)
- [x] EpochMetric (#2351)
- [x] Fbeta (#2340)
- [x] Loss (#2335)
- [x] MeanAbsoluteError (#2280)
- [x] MeanPairwiseDistance (#2307)
- [x] MeanSquaredError (#2280)
- [x] MetricsLambda (#2352)
- [x] MultiLabelConfusionMatrix (#2347)
- [x] Precision (#2340)
- [x] PSNR (#2311)
- [x] Recall (#2340)
- [x] RootMeanSquaredError (#2307)
- [x] RunningAverage (#2348)
- [x] SSIM (#2241)
- [x] TopKCategoricalAccuracy (#2284)
- [x] Bleu (#2317)
- [x] Rouge (#2317)
- [x] RougeL (#2317)
- [x] RougeN (#2317)
- [x] InceptionScore (#2349)
- [x] FID (#2349)

#### Contrib Metrics

- [x] AveragePrecision (#2341)
- [x] CohenKappa (#2321)
- [x] PrecisionRecallCurve (#2341)
- [x] ROC_AUC (#2341)
- [x] RocCurve (#2341)
- [x] CanberraMetric (#2323)
- [x] FractionalAbsoluteError (#2323)
- [x] FractionalBias (#2323)
- [x] GeometricMeanAbsoluteError (#2323)
- [x] GeometricMeanRelativeAbsoluteError (#2324)
- [x] ManhattanDistance (#2324)
- [x] MaximumAbsoluteError (#2324)
- [x] MeanAbsoluteRelativeError (#2324)
- [x] MeanError (#2324)
- [x] MeanNormalizedBias (#2324)
- [x] MedianAbsoluteError (#2324)
- [x] MedianAbsolutePercentageError (#2324) 
- [x] MedianRelativeAbsoluteError (#2324)
- [x] R2Score (#2324)
- [x] WaveHedgesDistance (#2324) 

This is a great issue for first comers and those who want to learn more about Sphinx. This issue is the community effort.
If you want to contribute, please comment on this issue with the metric you want to add doctest (one by one).

Example on how to add doctest, see #2241 

Thanks!

### PR DESCRIPTION
**Pull request by Eunjnnn**
Fixes #2265

Description: add doctest for Loss

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/loss.py [hunk 1]
```diff
@@ -42,40 +42,40 @@ class Loss(Metric):
...
 
-            import torch
-            import torch.nn as nn
-            from torch.nn.functional import nll_loss
-
-            from ignite.metrics import Accuracy, Loss
-            from ignite.engine import create_supervised_evaluator
-
-            model = ...
+            model = nn.Linear(10, 3)
```
Comment (sdesrozis) [inline ignite/metrics/loss.py:47]: Have a look to this [file](https://github.com/pytorch/ignite/blob/master/docs/source/conf.py). A default model is already defined and it would be great to use this one rather than define a new one.

File: ignite/metrics/loss.py [hunk 2]
```diff
@@ -42,40 +42,40 @@ class Loss(Metric):
...
 
             criterion = nll_loss
+            c_kwargs = {"reduction": "sum"}
 
-            metrics = {
-                "Accuracy": Accuracy(),
-                "Loss": Loss(criterion)
-            }
+            def process_function(engine, batch):
```
Comment (sdesrozis) [inline ignite/metrics/loss.py:52]: A default trainer is defined in this [file](https://github.com/pytorch/ignite/blob/master/docs/source/conf.py). You must use it. However, a specific setup is needed, please see the following example 

https://github.com/pytorch/ignite/blob/5455fa5885225f7d5e39aab31a796b7efc067768/ignite/handlers/param_scheduler.py#L364

**Comment by sdesrozis** [comment]
@Eunjnnn Thank you very much for your help ! I left some comments, tell me if you need help.

File: ignite/metrics/loss.py (hunk 2 above)
Comment (Eunjnnn) [reply ignite/metrics/loss.py:52]: i understand how to setup default trainer! 
But i can't get how/where to use default trainer in loss metrics

**Comment by sdesrozis** [comment]
@Eunjnnn Are you still working on this ? It would be great to merge soon.

File: ignite/metrics/loss.py (hunk 2 above)
Comment (Eunjnnn) [reply ignite/metrics/loss.py:56]: > 

can you give some advice on the question above?

Comment (sdesrozis) [reply ignite/metrics/loss.py:56]: Sorry I missed your question !!

This trainer should replace your engine. No longer need to have a process function too. Look the example I mentioned, there is no engine.

File: ignite/metrics/loss.py [hunk 3]
```diff
@@ -42,35 +42,32 @@ class Loss(Metric):
...
+        .. testcode::
 
-            from ignite.metrics import Accuracy, Loss
-            from ignite.engine import create_supervised_evaluator
-
-            model = ...
+            model = default_model
 
             criterion = nll_loss
+            c_kwargs = {"reduction": "sum"}
```
Comment (sdesrozis) [inline ignite/metrics/loss.py:54]: This is useless

File: ignite/metrics/loss.py [hunk 4]
```diff
@@ -42,35 +42,32 @@ class Loss(Metric):
...
 
-            metrics = {
-                "Accuracy": Accuracy(),
-                "Loss": Loss(criterion)
-            }
+            def output_transform(output):
```
Comment (sdesrozis) [inline ignite/metrics/loss.py:56]: Here, this is useless.

However, it could be great to add a very concise example showing how handle the multi output case mentioned in the doc, what do you think ?

File: ignite/metrics/loss.py [hunk 5]
```diff
@@ -42,35 +42,32 @@ class Loss(Metric):
...
+                y_pred, y = output
+                criterion_kwargs = c_kwargs
+                return y_pred, y, c_kwargs 
 
-            # global criterion kwargs
-            criterion_kwargs = {...}
+            metric = Loss(criterion, output_transform=output_transform)
```
Comment (sdesrozis) [inline ignite/metrics/loss.py:61]:
```suggestion
            metric = Loss(criterion)
```

File: ignite/metrics/loss.py [hunk 6]
```diff
@@ -42,35 +42,24 @@ class Loss(Metric):
         and the Loss metric using an ``evaluator`` created with
         :meth:`~ignite.engine.create_supervised_evaluator` method.
 
-        .. code-block:: python
+        .. testsetup:: *
+
```
Comment (sdesrozis) [inline ignite/metrics/loss.py:46]:
```suggestion

```

lint error, spaces should be removed.

**Comment by sdesrozis** [comment]
@Eunjnnn Thanks for the update. It’s almost done, please do the fix I suggested. Thanks again !

File: ignite/metrics/loss.py [hunk 7]
```diff
@@ -42,35 +42,24 @@ class Loss(Metric):
...
+            default_trainer = get_default_trainer()
 
-            from ignite.metrics import Accuracy, Loss
-            from ignite.engine import create_supervised_evaluator
-
-            model = ...
+        .. testcode::
 
+            model = default_model
             criterion = nll_loss
```
Comment (sdesrozis) [inline ignite/metrics/loss.py:52]:
```suggestion
            criterion = nn.NLLLoss()
```

File: docs/source/conf.py [hunk 8]
```diff
@@ -336,6 +336,7 @@ def run(self):
 
 import torch
 from torch import nn, optim
+from torch.nn.functional import nll_loss
```
Comment (sdesrozis) [inline docs/source/conf.py:339]:
```suggestion
```

See below, no need to introduce this import.

Comment (Eunjnnn) [reply docs/source/conf.py:339]: Oops! Thank you!!

**Review by sdesrozis — APPROVED** [review]
@Eunjnnn Thank you ! LGTM
